CMakeLists.txt

```diff
@@ -499,10 +499,20 @@ option(PG_DEBUG_REFS "PG Ref debugging is enabled" OFF)
 
 option(WITH_TESTS "enable the build of ceph-test package scripts/binaries" ON)
 
-option(WITH_FIO "Support for fio engines" OFF)
-if(WITH_FIO)
+# fio
+option(WITH_SYSTEM_FIO "require and build with system fio" OFF)
+set(HAVE_FIO TRUE)
+if(WITH_SYSTEM_FIO)
   find_package(fio REQUIRED)
-endif(WITH_FIO)
+  if(NOT FIO_FOUND)
```

Comment:
i think WITH_FIO would suffice here. the reason we want `HAVE_FOO` is that the C/C++ source or cmake script might want to check the existence of a certain library.

Reply:
@tchaikov - I removed WITH_FIO totally, assuming that we always build with fio enabled. New WITH_SYSTEM_FIO has been introduced to be able to switch between system and downloaded versions.. Given that is your comment still applicable?

Reply:
i am still in favor of keeping fio an option for user. as it is not always required from user's perspective. even for a ceph developer, it is optional.

Reply:
One of the rationales for this patch is to avoid periodic fio plugin build failures caused by incomplete code changes. Which in turn are caused by lack of automatic  fio plugin building. So I'd like to have it enabled by default. 
Are you OK with having an option to *disable* such a build rather than enable it to cover that case?

Reply:
ahh, i see now. how about disabling it by default, and enabling it in `run-make-check.sh`? so the fio plugin feature can be built by the jenkins' "make check" run?